### PR TITLE
Make AbsolutePath's private fields public.

### PR DESCRIPTION
**Pull request by willhite**
Also, add String() method to PathSpec.

### CODE REVIEW

File: go/spec/absolute_path.go [hunk 1]
```diff
@@ -18,9 +18,9 @@ import (
 var datasetCapturePrefixRe = regexp.MustCompile("^(" + datas.DatasetRe.String() + ")")
 
 type AbsolutePath struct {
-	dataset string
-	hash    hash.Hash
-	path    types.Path
+	Dataset string
```
Comment (arv) [inline go/spec/absolute_path.go]: GoDoc?

**Comment by kalman** [comment]
You should have @arv review I'm about to fly out

**Comment by willhite** [comment]
@arv PTAL

File: go/spec/absolute_path.go (hunk 1 above)
Comment (arv) [inline go/spec/absolute_path.go]: Don't you need comments on the fields for godoc to show things about them?

Comment (willhite) [inline go/spec/absolute_path.go]: I'm not sure what you mean. It seems like commenting the fields would be a little overkill on this and we have plenty of places where public types and/or their fields are not documented and they show up in godocs. Do we have a requirement that all public fields in public structs need doc? If so, I'll do it.

Comment (kalman) [inline go/spec/absolute_path.go]: I think we should comment them.

**Pull request merged**